### PR TITLE
Move cmd, internal, and pkg into a new Go module

### PR DESCRIPTION
**Pull request by abayer**
(this is an alternative approach to solving the same problem as #31 - this approach lets you still do `go get github.com/DataDog/rules_oci/go`, while #31 doesn't work with `go get ...`)

This lets us depend on `github.com/DataDog/rules_oci/go` in other projects without Gazelle getting confused by depending on the Go module `github.com/DataDog/rules_oci`, which Gazelle's update-repos turns into the label `@com_github_datadog_rules_oci`, and depending on the Bazel rules from this repo, which is named `@com_github_datadog_rules_oci`.

I've left the top-level go.mod for `release/main.go` (and `test/go-multiarch-image/main.go`, though since that doesn't import anything, it probably doesn't strictly need to be in a module), and added a new `go/repositories.bzl` to `WORKSPACE` and `release/BUILD.bazel` as the output for `//release:gazelle-update-repos`, using `go/go.mod`, with a new `//release:gazelle-update-repos-root` target for updating `repositories.bzl` using the root `go.mod`.

### CODE REVIEW

**Comment by abayer** [comment]
Sooooo this doesn't quite work right - Bazel and Gazelle get messy. If you run the new `bazel run //release:gazelle-update-repos-go`, to generate `go/repositories.bzl` with the dependencies in `go/go.mod`, and then run `bazel run //release:gazelle-update-repos`, to generate `./repositories.bzl` with the dependencies in `./go.mod`, `go/repositories.bzl` gets wiped out. I'm trying to find how exactly you're supposed to generate a `repositories.bzl` for a repo with multiple `go.mod` files and a single `WORKSPACE`, but am having little luck so far.

https://github.com/bazelbuild/bazel-gazelle/issues/634 exists but is closed as won't-fix, so...yeah. More exploring to do, I guess.

**Comment by abayer** [comment]
Yaaay - moving out of draft. I've tested this with `go get github.com/abayer/rules_oci/go@451db1aabcf4eef179f90f2cff3115f5883d3e7e` (which burps a bit because the module is still `github.com/DataDog/rules_oci/go`, but that's only an issue because I'm on a fork, and `replace` deals with that), and it works. Then I ran `gazelle --update-repos ...` and it correctly created a `go_repository` for the `importpath` `github.com/DataDog/rules_oci/go` (with a replace to point to my fork, of course), etc. So yeah, this works!

File: README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ manifests, image indexes (multi-arch images) and custom artifacts
 * **Multi-arch images**, compiling and building multi-arch images with a single Bazel invocation
 
 In addition to Bazel rules, we offer many helpers for interacting with OCI
-artifacts under the `pkg` directory and a CLI tool for creating new OCI
+artifacts under the `go/pkg` directory and a CLI tool for creating new OCI
```
Comment (jprobinson) [inline README.md:15]: May be worth noting here that this is also in its own Go module and what to "go get" if they do want to use the helpers.

**Review by jprobinson — APPROVED** [review]
📦 🌯

File: README.md (hunk 1 above)
Comment (abayer) [reply README.md:15]: Good call, will add that momentarily.

Comment (abayer) [reply README.md:15]: Done!

**Comment by guyboltonking** [comment]
@abayer we have collided; I have a handful of PRs I'd like to merge (#33, #34); what order should we do these in?

**Comment by abayer** [comment]
@guyboltonking - feel free to go first! It'll be easy enough for me to rebase.

**Pull request merged**